src/lpython/semantics/python_ast_to_asr.cpp

```diff
@@ -2923,10 +2932,109 @@ class CommonVisitor : public AST::BaseVisitor<Struct> {
         assign_asr_target = assign_asr_target_copy;
     }
 
+    void visit_Constructor(const AST::FunctionDef_t &x,
+        Vec<char*>& member_names, Vec<ASR::call_arg_t> &member_init){
+        if(x.n_decorator_list>0){
+            throw SemanticError("Decorators for __init__ not implemented" , x.base.base.loc);
+        }
+        if( x.m_args.n_args > 1 ){
+            throw SemanticError("Only default constructors implemented " , x.base.base.loc);
+        }
+        std::string self_name = x.m_args.m_args[0].m_arg;
+        for(size_t i = 0; i < x.n_body; i++){
+            std::string var_name;
+            if (! AST::is_a<AST::AnnAssign_t>(*x.m_body[i]) ){
+                throw SemanticError("Only AnnAssign implemented in __init__ " , x.m_body[i]->base.loc);
+            }
+            AST::AnnAssign_t ann_assign = *AST::down_cast<AST::AnnAssign_t>(x.m_body[i]);
+            if(AST::is_a<AST::Attribute_t>(*ann_assign.m_target)){
+                AST::Attribute_t* a = AST::down_cast<AST::Attribute_t>(ann_assign.m_target);
+                if(AST::is_a<AST::Name_t>(*a->m_value)){
+                    AST::Name_t* n = AST::down_cast<AST::Name_t>(a->m_value);
+                    if(std::string(n->m_id)!=self_name && current_scope->resolve_symbol(std::string(n->m_id))){
+                        throw SemanticError("Symbol not declared" , x.m_body[i]->base.loc);
```

Comment:
So, we do not add self to the SymbolTable, right?
I think we need to add it to the symbol table. So, the objects will be access through self.

Reply:
Also, the type of self will be StructType, so for that, I think we need to declare an empty Class first and then visit the body. That way the class symbol would be available in the scope.

Reply:
The `StructType` requires all the members and their types to be added to it. I am not sure if it will work to declare the type of `self` as the `StructType`.